test/e2e/framework/.import-restrictions

```diff
@@ -11,14 +11,49 @@ rules:
   - selectorRegexp: ^[a-z]+(/|$)
     allowedPrefixes: [ "" ]
 
-  # Ginkgo + Gomega.
+  # stdlib x and proto
+  - selectorRegexp: ^golang.org/x|^google.golang.org/protobuf
+    allowedPrefixes: [ "" ]
+
+  # Ginkgo + Gomega
   - selectorRegexp: github.com/onsi/(ginkgo|gomega)|^k8s[.]io/kubernetes/test/(e2e/framework/internal/|utils)
     allowedPrefixes: [ "" ]
 
+  # kube-openapi
+  - selectorRegexp: ^k8s.io/kube-openapi
+    allowedPrefixes: [ "" ]
+
+  # Public SIG Repos
+  - selectorRegexp: ^sigs.k8s.io/(json|yaml|structured-merge-diff)
+    allowedPrefixes: [ "" ]
+
   # some of the shared test helpers (but not E2E sub-packages!)
   - selectorRegexp: ^k8s[.]io/kubernetes/test/(e2e/framework/internal/|utils)
     allowedPrefixes: [ "" ]
 
+  # Third party deps
+  - selectorRegexp: ^github.com/|^gopkg.in
+    allowedPrefixes: [ 
+      "gopkg.in/inf.v0",
+      "github.com/davecgh/go-spew/spew",
+      "github.com/evanphx/json-patch",
+      "github.com/go-logr/logr",
+      "github.com/gogo/protobuf/proto",
+      "github.com/gogo/protobuf/sortkeys",
+      "github.com/golang/protobuf/proto",
+      "github.com/google/gnostic-models/openapiv2",
+      "github.com/google/gnostic-models/openapiv3",
+      "github.com/google/go-cmp/cmp",
+      "github.com/google/go-cmp/cmp/cmpopts",
+      "github.com/google/gofuzz",
+      "github.com/google/uuid",
+      "github.com/imdario/mergo",
+      "github.com/spf13/cobra",
+      "github.com/spf13/pflag",
+      "github.com/stretchr/testify/assert",
+      "github.com/stretchr/testify/require"
```

Comment:
this list is awkward. I vote for reverting in gengo. This breaks import-boss user experience IMO.

Reply:
https://github.com/kubernetes/gengo/pull/236 doesn't say exactly what was fixed. It looks like transitive dependencies are now also checked, not just direct dependencies.

I can see arguments both ways. If I only check direct dependencies, then I trust the maintainers of those other packages to not pull in something nasty. My own config then is smaller and doesn't need to be updated when updating some dependency.

But is that trust really justified? I suspect not. Other maintainers might have different criteria, so if I want to be thorough, I have to check everything.

Reply:
We just shouldn't have to list external packages here. Import-boss is about kube packages. If we require such a list, ppl will start just adding whatever the verify CI job complains about. It's a little like the broken window effect. In all the exceptions it is hard to see why not add yet another one.

Reply:
> this list is awkward. I vote for reverting in gengo. This breaks import-boss user experience IMO.


> It looks like transitive dependencies are now also checked, not just direct dependencies

The change in #236 I don't think enabled the checking of transitive dependencies.

import-boss checks imports of all input packages by filtering out all packages in the gengo `Universe` that weren't directly specified on the command line:

https://github.com/kubernetes/gengo/blob/9cce18d56c01fab1d8ed72411ad6ef7005fc33a4/examples/import-boss/generators/import_restrict.go#L60-L70

#236 fixed a bug where files genuinely part of the command-line input were being skipped by import-boss. I don't think it would be correct to revert it.

This is the new code that was added to `inputIncludes` by #236:

https://github.com/kubernetes/gengo/blob/9cce18d56c01fab1d8ed72411ad6ef7005fc33a4/examples/import-boss/generators/import_restrict.go#L106-L111

Most inputs we use in our invocation end in `...`

https://github.com/kubernetes/kubernetes/blob/c1628f3b6f72b5666058e24175decdaffb09fdac/hack/verify-import-boss.sh#L33-L45

Prior to #236 inputIncludes([]string{"k8s.io/cloud-provider/..."}, "k8s.io/cloud-provider/config") would return `false`, and `config` package would not be checked.

Reply:
Ack, makes sense.

I didn't realize that also before we were specifying non-k8s packages here. I was of the impression that long ago this was different.